src/sentry/static/sentry/app/components/modals/teamAccessRequestModal.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
+import {Client} from 'app/api';
+import {ModalRenderProps, TeamAccessRequestModalOptions} from 'app/actionCreators/modal';
+import {t} from 'app/locale';
+import Button from 'app/components/button';
+import space from 'app/styles/space';
+import withApi from 'app/utils/withApi';
+
+type Props = ModalRenderProps &
+  TeamAccessRequestModalOptions & {
+    api: Client;
+    memberId: string;
+    teamId: string;
+    orgId: string;
+  };
+
+type State = {
+  createBusy: boolean;
+};
+
+class CreateTeamAccessRequest extends React.Component<Props, State> {
+  state: State = {
+    createBusy: false,
+  };
+
+  handleClick = async () => {
+    const {api, memberId, orgId, teamId, closeModal} = this.props;
+
+    this.setState({createBusy: true});
+
+    try {
+      await api.requestPromise(
+        `/organizations/${orgId}/members/${memberId}/teams/${teamId}/`,
+        {
+          method: 'POST',
+        }
+      );
+      this.setState({createBusy: false});
+      addSuccessMessage(t('Team request sent for approval'));
+    } catch (err) {
+      this.setState({createBusy: false});
```

Comment:
I think you can pull this setState out to after the try/catch and just have it once.